Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs

```diff
@@ -24,144 +22,84 @@ public partial class TapToPlace : MonoBehaviour
-            else
-            {
-                Debug.Log(gameObject.name + " : " + "World anchor loaded from anchor store and updated for this game object.");
-            }
-        }
 
-        // Called by GazeGestureManager when the user performs a tap gesture.
-        void OnSelect()
-        {
-            if (SpatialMappingManager.Instance != null)
+            spatialMappingManager = SpatialMappingManager.Instance;
+            if (spatialMappingManager == null)
             {
-                // On each tap gesture, toggle whether the user is in placing mode.
-                placing = !placing;
-
-                // If the user is in placing mode, display the spatial mapping mesh.
-                if (placing)
-                {
-                    SpatialMappingManager.Instance.DrawVisualMeshes = true;
-
-                    Debug.Log(gameObject.name + " : " + "Removing existing world anchor if any.");
-
-                    // Remove existing world anchor when moving an object.
-                    DestroyImmediate(gameObject.GetComponent<WorldAnchor>());
-
-                    // Delete existing world anchor from anchor store when moving an object.
-                    if (anchorStore != null)
-                    {
-                        anchorStore.Delete(SavedAnchorFriendlyName);
-                    }
-                }
-                // If the user is not in placing mode, hide the spatial mapping mesh.
-                else
-                {
-                    SpatialMappingManager.Instance.DrawVisualMeshes = false;
-
-                    // Add world anchor when object placement is done.
-                    CreateAnchor();
-                }
-            }
-            else
-            {
-                Debug.Log("TapToPlace requires spatial mapping.  Try adding SpatialMapping prefab to project.");
+                Debug.LogError("This script expects that you have a SpatialMappingManager component in your scene.");
             }
-        }
-
-        private void CreateAnchor()
-        {
-            // NOTE: It's good practice to ensure your parent hierarchy or root game object does not already have a World Anchor.
-            // You can handle this in a way that works best for your application.
-            // For example: gameObject.transform.root.GetComponent<WorldAnchor>();
 
-            // Add the world anchor component when done moving an object.
-            WorldAnchor anchor = gameObject.AddComponent<WorldAnchor>();
-            if (anchor.isLocated)
+            if (anchorManager != null && spatialMappingManager != null)
             {
-                SaveAnchor(anchor);
+                anchorManager.AttachAnchor(this.gameObject, SavedAnchorFriendlyName);
             }
             else
             {
-                anchor.OnTrackingChanged += Anchor_OnTrackingChanged;
+                // If we don't have what we need to proceed, we may as well remove ourselves.
+                Destroy(this);
```

Comment:
This seems wrong. Why are we removing TapToPlace from the gameobject? Did I read this wrong?